.tekton/build_and_trigger.yaml

```diff
@@ -85,56 +112,45 @@ spec:
-          - name: BUILDER_IMAGE
-            value: $(params.builder-image)
-          - name: DOCKERFILE
-            value: $(workspaces.source.path)/$(params.dockerfile)
         runAfter:
           - calculate-tag
         taskRef:
           name: buildah
           kind: ClusterTask
-      - name: trigger-tests-and-push
-        computeResources:
-          requests:
-            cpu: 100mi
-        params:
-          - name: revision
-            value: "{{ revision }}"
-          - name: event_type
-            value: "{{ event_type }}"
-          - name: image_tag
-            value: $(tasks.calculate-tag.results.image_tag)
+      - name: test-clair
         runAfter:
           - build-image
         taskSpec:
-          params:
-            - name: revision
-              type: string
-            - name: event_type
-              type: string
-            - name: image_tag
-              type: string
           steps:
-            - name: tigger-workflow
-              image: registry.access.redhat.com/ubi9/python-39
-              env:
-                - name: GITHUB_TOKEN
-                  valueFrom:
-                    secretKeyRef:
-                      name: clair-in-ci-db-github-token
-                      key: github-token
+            - name: test-clair-version
+              image: http://quay.io/redhat-appstudio/clair-in-ci:$(tasks.calculate-tag.results.image_tag)
               script: |
                 #!/usr/bin/env bash
-                event="pull_request"
-                if [ "$(params.event_type)" == "push" ]; then
-                  event="push"
-                fi
-                echo "Acting according to event: "$event""
-                curl --fail-with-body -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $GITHUB_TOKEN" --request POST --data '{"event_type":"trigger_workflow", "client_payload": {"pr_event": "'"$event"'", "image_tag": "$(params.image_tag)"}}' https://api.github.com/repos/konflux-ci/clair-in-ci-db/dispatches
+                echo "See current version of Clair below:"
+                clair-action --version
+            - name: get-clair-output
+              image: http://quay.io/redhat-appstudio/clair-in-ci:$(tasks.calculate-tag.results.image_tag)
+              script: |
+                #!/usr/bin/env bash
+                echo "Test real life usage of Clair"
+                clair-action report --image-ref=registry.access.redhat.com/ubi9-minimal --db-path=/tmp/matcher.db --format=quay > /tekton/home/clairdata.json
+            - name: check-clair-output
+              image: quay.io/redhat-appstudio/konflux-test:latest
+              script: |
+                #!/usr/bin/env bash
+                echo "Check if the output is as expected"
+                jq -e '.data[].Features[0] | select(has("Name") and has("Vulnerabilities")) or error("Required keys do not exist")' /tekton/home/clairdata.json
+      - name: login-and-push
```

Comment:
you want to run this step only for push event right, not for each pullrequest

Reply:
@MartinBasti I am going to resolve this so that I can run the checks, it can be reopened.

Reply:
I think this can be addressed now, this task should have a `when` clause to not run on PRs.

Reply:
@dirgim I put in this line at the time of the original comment 

if [ "$(params.event_type)" != "pull_request" ]; then

If you prefer a different approach thats no problem

Reply:
Ah, I missed that.
I still think it would be more visible and maintainable as a `when` clause (here's an [example from build pipelines](https://github.com/konflux-ci/build-definitions/blob/7c58b730e89d819bf5dc7e5a8c1bdd82f1ad6708/pipelines/template-build/template-build.yaml#L73-L76)).
Also I think that this task should only run when event_type is `push` because other event types may occur during a pull request - e.g. when commenting `/retest` the event type is `retest-all-comment`.

Reply:
thanks :)